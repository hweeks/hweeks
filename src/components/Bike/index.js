import React from 'react';
import PropTypes from 'prop-types';
import {
  BikeContainer,
  BikeImage,
  BikeName,
  BikeInfo
} from './styles'

class Bike extends React.PureComponent {
  buildImageInfo() {
    const imageObject = {};
    imageObject.alt = `A picture of a ${this.props.name} bicycle.`;
    imageObject.src = `assets/img/${this.props.imagename}_small.jpg`;
    imageObject.srcset = `assets/img/${this.props.imagename}_large.jpg 1024w, assets/img/${
      this.props.imagename}_medium.jpg 640w, assets/img/${
      this.props.imagename}_small.jpg 320w`;
    return imageObject;
  }
  render() {
    const imageObject = this.buildImageInfo();
    return (
      <BikeContainer>
        <BikeImage
          src={imageObject.src}
          srcSet={imageObject.srcset}
          alt={imageObject.alt}
        />
        <BikeName>{this.props.name}</BikeName>
        <BikeInfo>
          size: {this.props.size}cm
        </BikeInfo>
      </BikeContainer>
    );
  }
}

Bike.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  imagename: PropTypes.string.isRequired,
};

export default Bike;
