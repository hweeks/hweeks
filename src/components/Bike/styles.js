import styled from 'styled-components';

export const BikeContainer = styled.div`
  display: flex;
  flex-flow: column;
  border-bottom: ${props => props.theme.borderDefault};
  border-color: ${props => props.theme.syntax.constant};
  padding: ${
  props => props.theme.atom * 2
}rem ${
  props => props.theme.atom
}rem;

  &:last-child {
    border: 0;
  }
`;

export const BikeImage = styled.img`
  height: auto;
  max-width: 100%;
  padding: ${props => props.theme.atom * 3}rem 0;
  @media (min-width: ${props => props.theme.breakpoint}rem) {
    order: 3;
  }
`;

export const BikeName = styled.p`
  font-size: ${props => props.theme.headerFontSize}rem;
  color: ${props => props.theme.syntax.tag};
  @media (min-width: ${props => props.theme.breakpoint}rem) {
    order: 1;
  }
`;

export const BikeInfo = styled.p`
  font-size: ${props => props.theme.bodyFontSize}rem;
  color: ${props => props.theme.syntax.entity};
  @media (min-width: ${props => props.theme.breakpoint}rem) {
    order: 2;
  }
`;
