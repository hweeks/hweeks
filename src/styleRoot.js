const rootValues = {
  atom: '0.4',
  headerFontSize: '2.2',
  bodyFontSize: '1.6',
  borderColor: '#666',
  bgColor: '#fff',
  borderDefault: `0.2rem solid #666`,
  breakpoint: '41.5',
}

export default rootValues
