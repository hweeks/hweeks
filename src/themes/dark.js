export const dark = {
  common: {
    accent: '#E6B450',
    bg: '#0A0E14',
    fg: '#B3B1AD',
    ui: '#3D424D',
  },
  syntax: {
    tag: '#39BAE6',
    func: '#FFB454',
    entity: '#59C2FF',
    string: '#C2D94C',
    regexp: '#95E6CB',
    markup: '#F07178',
    keyword: '#FF8F40',
    special: '#E6B673',
    comment: '#626A73',
    constant: '#FFEE99',
    operator: '#F29668',
    error: '#FF3333',
  },
  ui: {
    line: '#01060E',
    panel: {
      bg: '#0D1016',
      shadow: '#00010A',
      border: '#000000',
    },
    gutter: {
      normal: '#3D424D99',
      active: '#3D424DE6',
    },
    selection: {
      bg: '#273747',
      inactive: '#1F2C39',
      border: '#304357',
    },
    guide: {
      active: '#3D424DB3',
      normal: '#3D424D66',
    },
  },
  vcs: {
    added: '#91B362',
    modified: '#6994BF',
    removed: '#D96C75',
  },
};
