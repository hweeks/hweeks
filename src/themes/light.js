export const light = {
  common: {
    accent: '#FF9940',
    bg: '#FAFAFA',
    fg: '#6C7680',
    ui: '#959DA6',
  },
  syntax: {
    tag: '#55B4D4',
    func: '#F2AE49',
    entity: '#399EE6',
    string: '#86B300',
    regexp: '#4CBF99',
    markup: '#F07171',
    keyword: '#FA8D3E',
    special: '#E6BA7E',
    comment: '#ABB0B6',
    constant: '#A37ACC',
    operator: '#ED9366',
    error: '#F51818',
  },
  ui: {
    line: '#959DA61A',
    panel: {
      bg: '#FFFFFF',
      shadow: '#56606940',
      border: '#F0F0F0',
    },
    gutter: {
      normal: '#959DA666',
      active: '#959DA6CC',
    },
    selection: {
      bg: '#EDF0F5',
      inactive: '#F2F4F7',
      border: '#E5EBF2',
    },
    guide: {
      active: '#959DA6B3',
      normal: '#959DA64D',
    },
  },
  vcs: {
    added: '#99BF4D',
    modified: '#709ECC',
    removed: '#F27983',
  },
};
