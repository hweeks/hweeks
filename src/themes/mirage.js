export const mirage = {
  common: {
    accent: '#FFCC66',
    bg: '#1F2430',
    fg: '#CBCCC6',
    ui: '#707A8C',
  },
  syntax: {
    tag: '#5CCFE6',
    func: '#FFD580',
    entity: '#73D0FF',
    string: '#BAE67E',
    regexp: '#95E6CB',
    markup: '#F28779',
    keyword: '#FFA759',
    special: '#FFE6B3',
    comment: '#5C6773',
    constant: '#D4BFFF',
    operator: '#F29E74',
    error: '#FF3333',
  },
  ui: {
    line: '#191E2A',
    panel: {
      bg: '#232834',
      shadow: '#141925',
      border: '#101521',
    },
    gutter: {
      normal: '#707A8C66',
      active: '#707A8CCC',
    },
    selection: {
      bg: '#34455A',
      inactive: '#2D3B4D',
      border: '#3C526A',
    },
    guide: {
      active: '#707A8CB3',
      normal: '#707A8C4D',
    },
  },
  vcs: {
    added: '#A6CC70',
    modified: '#77A8D9',
    removed: '#F27983',
  },
};
