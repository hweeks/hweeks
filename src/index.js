import ReactDOM from 'react-dom';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import styled, { ThemeProvider } from 'styled-components';
import App from './composite/App';
import * as theme from './themes';

const ThemeSelect = styled.div`
  display: ${props => (props.display ? 'block' : 'none')};
`;

const ThemeSelectWrapper = styled.div`
  position: fixed;
  bottom: 0;
`;

class AppWrapper extends React.PureComponent {
  constructor(props) {
    super(props);
    const themes = ['dark', 'light', 'mirage'];
    this.handleThemeSelect = this.handleThemeSelect.bind(this);
    this.toggleThemer = this.toggleThemer.bind(this);
    this.state = {
      theme: theme[themes[Math.floor(Math.random() * 3) % 3]],
      menuShown: false,
    };
  }
  handleThemeSelect({ target }) {
    if (theme[target.value]) {
      this.setState({
        theme: theme[target.value],
        menuShown: false,
      });
    }
  }
  toggleThemer() {
    this.setState({
      menuShown: !this.state.menuShown,
    });
  }
  render() {
    return (
      <BrowserRouter>
        <ThemeProvider theme={this.state.theme}>
          <React.Fragment>
            <App />
            <ThemeSelectWrapper>
              <button onClick={this.toggleThemer}>
                theme
              </button>
              <ThemeSelect display={this.state.menuShown}>
                <button onClick={this.handleThemeSelect} value="light">light</button>
                <button onClick={this.handleThemeSelect} value="dark">dark</button>
                <button onClick={this.handleThemeSelect} value="mirage">mirage</button>
              </ThemeSelect>
            </ThemeSelectWrapper>
          </React.Fragment>
        </ThemeProvider>
      </BrowserRouter>
    );
  }
}

const load = () => {
  ReactDOM.render(
    <AppWrapper />,
    document.querySelector('#home'),
  );
};

if (document.readyState !== 'complete') {
  document.addEventListener('DOMContentLoaded', load);
} else {
  load();
}
