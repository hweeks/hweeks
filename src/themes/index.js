import { light as lTheme } from './light';
import { mirage as mTheme } from './mirage';
import { dark as dTheme } from './dark';

const rootValues = {
  atom: '0.4',
  headerFontSize: '2.2',
  bodyFontSize: '1.6',
  borderColor: '#666',
  bgColor: '#fff',
  borderDefault: '0.2rem solid #666',
  breakpoint: '41.5',
};

export const light = { ...rootValues, ...lTheme };

export const dark = { ...rootValues, ...dTheme };

export const mirage = { ...rootValues, ...mTheme };
