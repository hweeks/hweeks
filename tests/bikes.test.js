import React from 'react';
import renderer from 'react-test-renderer';
import Bike from '../src/components/Bike';
import { ThemeProvider } from 'styled-components'
import {light} from '../src/themes'

const bikeSettings = {
  name: 'bike name',
  imagename: 'bmc',
  size: 20,
};

test('Bike actually renders', () => {
  const component = renderer.create(
    <ThemeProvider theme={light}>
      <Bike
        imagename={bikeSettings.imagename}
        name={bikeSettings.name}
        size={bikeSettings.size}
      />
    </ThemeProvider>
    ,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
