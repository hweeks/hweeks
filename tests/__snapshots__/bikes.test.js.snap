// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bike actually renders 1`] = `
<div
  className="sc-bdVaJa uVrdD"
>
  <img
    alt="A picture of a bike name bicycle."
    className="sc-bwzfXH cmMosU"
    src="assets/img/bmc_small.jpg"
    srcSet="assets/img/bmc_large.jpg 1024w, assets/img/bmc_medium.jpg 640w, assets/img/bmc_small.jpg 320w"
  />
  <p
    className="sc-htpNat kPwbpi"
  >
    bike name
  </p>
  <p
    className="sc-bxivhb dTggbt"
  >
    size: 
    20
    cm
  </p>
</div>
`;
